### ISSUE TITLE
[BUG] langchain4j-spring: Application crashes if RetrievalAugmentor is defined

### ISSUE THREAD

**Issue by vprudnikov** [langchain4j/langchain4j#1036]
**Describe the bug**
Application crashes if RetrievalAugmentor is defined

**Log and Stack trace**
```
Caused by: dev.langchain4j.exception.IllegalConfigurationException: Only one out of [retriever, contentRetriever, retrievalAugmentor] can be set
	at dev.langchain4j.exception.IllegalConfigurationException.illegalConfiguration(IllegalConfigurationException.java:12) ~[langchain4j-0.30.0.jar:na]
	at dev.langchain4j.service.AiServices.retrievalAugmentor(AiServices.java:393) ~[langchain4j-0.30.0.jar:na]
	at dev.langchain4j.service.spring.AiServiceFactory.getObject(AiServiceFactory.java:85) ~[langchain4j-spring-boot-starter-0.30.0.jar:na]
	at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.doGetObjectFromFactoryBean(FactoryBeanRegistrySupport.java:182) ~[spring-beans-6.1.6.jar:6.1.6]
	... 43 common frames omitted
```

**To Reproduce**

Define a RetrievalAugmentor bean, without ContentRetriever, like so:
```
@Bean
RetrievalAugmentor retrievalAugmentor(...) {
   return DefaultRetrievalAugmentor.builder()
                 ....
                 .build();
}
```

**Expected behavior**
The application starts successfully.

**Please complete the following information:**
- LangChain4j version: 0.30.0
- LLM(s) used: N/A
- Java version: 21
- Spring Boot version (if applicable): 3.2.5

**Additional context**
I assume the problem is with the `RagAutoConfig` auto-configuration:
```
   @Bean
    @ConditionalOnBean({
            EmbeddingModel.class,
            EmbeddingStore.class
    })
    @ConditionalOnMissingBean
    ContentRetriever contentRetriever(EmbeddingModel embeddingModel, ...
```
As you can see, the ContentRetriever is ALWAYS registered, whether RetrievalAugmentor is already defined or not. 
I think the fix could be adding both ContentRetriever and RetrievalAugmentor to the @ConditionalOnMissingBean annotation.

**However**, at the same time, I wouldn't be so strict and I would change the behavior by saying the `RetrievalAugmentor` always takes priority. It would work better (for me at least) because I could define my own `RetrievalAugmentor` like so:
```
@Bean
RetrievalAugmentor retrievalAugmentor(ContentRetriever contentRetriever...)
```
In the above example, my RetrievalAugmentor expects an existing bean (either automatically created by the auto-configuration or defined by me).

**Comment by langchain4j** [comment]
HI @vprudnikov please check https://github.com/langchain4j/langchain4j/issues/887, it has been fixed in 0.30.1 (only spring modules).

**Comment by vprudnikov** [comment]
thanks, @langchain4j ! I confirm that it has been fixed in 0.30.1.
